.github/workflows/build-test.yaml

```diff
@@ -13,14 +14,29 @@ concurrency:
 jobs:
   can-run-ci:
     runs-on: ubuntu-20.04
-    # This workflow trigger may lead to malicious PR authors being able to obtain repository write permissions or stealing repository secrets.
+    # if the event is pull_request:
+    #   - this is not a fork
+    #   - and not dependabot
+    # if the event is pull_request_target:
+    #   - this is dependabot
+    #   - or this is a fork and has label '@actions/safe-to-test'
+    #
+    # The 'pull_request_target' workflow trigger may lead to malicious PR authors being able to obtain repository write permissions or stealing repository secrets.
     # Please read https://securitylab.github.com/research/github-actions-preventing-pwn-requests/
-    # only run this workflow if:
-    #   not a fork or user is dependabot or PR is labeled with '@actions/safe-to-test'
     if: >
-      github.event.pull_request.head.repo.full_name == github.repository ||
-      github.event.pull_request.user.login == 'dependabot[bot]' ||
-      contains(github.event.pull_request.labels.*.name, '@actions/safe-to-test')
+      ( github.event_name == 'pull_request' &&
+        ( github.event.pull_request.head.repo.full_name == github.repository &&
+          github.event.pull_request.user.login != 'dependabot[bot]'
+        )
+      )
+      ||
+      ( github.event_name == 'pull_request_target' && 
+        ( github.event.pull_request.user.login == 'dependabot[bot]' ||
```

Comment:
what if someone is using dependabot on their fork?

Reply:
i guess my question is: is `dependabot[bot]` user/username unique across all of github?

Reply:
it wouldnt make a dependabot pr into the origin. it would into their fork. i suppose in that case the workflow would run on their fork but they wouldnt have secrets anyways and it would fail.